# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
# Commit: 709078257c62
# Full Hash: 709078257c628fae054382894de5864b1f5f5c2a
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 3
# Description:
#   Bug 1654112 - Re-wire GMP PluginID between codecs and VideoConduit. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104724
# ==============================================================================

diff -r 3ca24e761cc3 -r 709078257c62 dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp	Wed Feb 10 10:34:14 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp	Wed Feb 10 15:17:29 2021 +0100
@@ -333,6 +333,23 @@
       mRecvSSRC(0),
       mRemoteSSRC(0) {
   mRecvStreamConfig.renderer = this;
+
+  mSendPluginCreated = mEncoderFactory->CreatedGmpPluginEvent().Connect(
+      GetMainThreadSerialEventTarget(), [this](uint64_t aPluginID) {
+        mSendCodecPluginIDs.AppendElement(aPluginID);
+      });
+  mSendPluginReleased = mEncoderFactory->ReleasedGmpPluginEvent().Connect(
+      GetMainThreadSerialEventTarget(), [this](uint64_t aPluginID) {
+        mSendCodecPluginIDs.RemoveElement(aPluginID);
+      });
+  mRecvPluginCreated = mDecoderFactory->CreatedGmpPluginEvent().Connect(
+      GetMainThreadSerialEventTarget(), [this](uint64_t aPluginID) {
+        mRecvCodecPluginIDs.AppendElement(aPluginID);
+      });
+  mRecvPluginReleased = mDecoderFactory->ReleasedGmpPluginEvent().Connect(
+      GetMainThreadSerialEventTarget(), [this](uint64_t aPluginID) {
+        mRecvCodecPluginIDs.RemoveElement(aPluginID);
+      });
 }
 
 WebrtcVideoConduit::~WebrtcVideoConduit() {
@@ -1000,6 +1017,10 @@
   }
 
   mCall->UnregisterConduit(this);
+  mSendPluginCreated.Disconnect();
+  mSendPluginReleased.Disconnect();
+  mRecvPluginCreated.Disconnect();
+  mRecvPluginReleased.Disconnect();
   {
     MutexAutoLock lock(mMutex);
     DeleteSendStream();